arch/riscv32/core/fatal.c

```diff
@@ -199,8 +199,9 @@ FUNC_NORETURN void _Fault(const NANO_ESF *esf)
 	__asm__ volatile("csrr %0, mcause" : "=r" (mcause));
 
 	mcause &= SOC_MCAUSE_EXP_MASK;
-
+#ifdef CONFIG_PRINTK
 	printk("Exception cause %s (%d)\n", cause_str(mcause), (int)mcause);
+#endif
```

Comment:
Users don't expect to see ifdef's around CONFIG_PRINTK.

We should instead drop the #ifdef CONFIG_PRINTK around mcause. This has no performance impact, and makes this function easier to read.

Reply:
This was merged as part of #11200. I will add a commit making this change to this series.